app/models/pool.rb

```diff
@@ -0,0 +1,53 @@
+class Pool < ApplicationRecord
+  belongs_to :event
```

Comment:
I don't remember how but I managed by testing the app on this branch to create a pool with size 0.

We should add some validation to avoid this scenario.

```ruby
>> self.sell.events_for_year(2022).second.capital_gain
=> NaN
>> self.sell.events_for_year(2022).second
=> #<Event id: 2, action: "sell", user_id: 1, currency_id: 1, stock_id: 1, quantity: 1, price: 0.1e1, commission: 0.0, executed_on: "2022-01-15", created_at: "2022-01-03 09:22:02.558546000 +0000", updated_at: "2022-01-03 09:22:02.558546000 +0000", sort_column: "2022-01-15/1/1641201722.554402">
>> self.sell.events_for_year(2022).second.pool
=> #<Pool id: 297, event_id: 2, value: 0.0, size: 0>
```

Reply:
This would happen if `previous_event.pool.size` is zero. But how did you end up with an empty pool for two events in a row? The last event sells 1 stock, so the previous pool size should be 1.

I hope you can come up with the reproduction steps for this issue. You can send me e.g. `development.log` or your database file on email or Slack maybe.

Reply:
```
irb(main):021:0> Pool.delete_all
  Pool Delete All (3.0ms)  DELETE FROM "pools"
=> 3
irb(main):022:0> Event.delete_all
  Event Delete All (1.2ms)  DELETE FROM "events"
=> 3

### Create from web a sell event as first
Event Create (0.4ms)  INSERT INTO "events" ("action", "user_id", "currency_id", "stock_id", "quantity", "price", "commission", "executed_on", "created_at", "updated_at", "sort_column") VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)  [["action", 1], ["user_id", 1], ["currency_id", 1], ["stock_id", 1], ["quantity", 100], ["price", 10.0], ["commission", 0.0], ["executed_on", "2022-01-15"], ["created_at", "2022-01-03 11:46:50.612654"], ["updated_at", "2022-01-03 11:46:50.612654"], ["sort_column", "2022-01-15/1/1641210410.607897"]]
  ↳ app/controllers/events_controller.rb:42:in `create'

  Pool Create (0.4ms)  INSERT INTO "pools" ("event_id", "value", "size") VALUES (?, ?, ?)  [["event_id", 1], ["value", 0.0], ["size", 0]]
  ↳ app/models/pool.rb:33:in `create_event'


irb(main):023:0> Pool.last
  Pool Load (0.1ms)  SELECT "pools".* FROM "pools" ORDER BY "pools"."id" DESC LIMIT ?  [["LIMIT", 1]]
=> #<Pool:0x00007fea329cd988 id: 298, event_id: 1, value: 0.0, size: 0>

```

Reply:
Also try to add and destroy an event. That creates an exception
```
ActiveRecord::RecordNotFound at /events/1
Couldn't find Event without an ID

in app/models/pool.rb, line 10
   # (Recursively) add a Pool for this event and all after it
    last_event = Event.find(later_event_ids.last)
    create_event(last_event)

```

Reply:
In real life, the first event cannot be a sell. You need to have something in the pool before you can sell from it.

And yes, probably deleting the only event in a pool causes problems. I’ll have a look.

Reply:
I pushed a commit that fixes at least the "add and destroy an event" problem.